apps/prairielearn/assets/scripts/instructorAssessmentInstancesClient.ts

```diff
@@ -0,0 +1,739 @@
+          </div>
+          <div
+            class="progress-bar bg-danger"
+            style="width: ${100 - Math.floor(Math.min(100, score))}%"
+          >
+            ${score < 50 ? `${Math.floor(score)}%` : ''}
+          </div>
+        </div>
+      `;
+    } else {
+      return '';
+    }
+  }
+
+  function listFormatter(list: string[]) {
+    if (!list || !list[0]) list = ['(empty)'];
+    return html`<small>${list.join(', ')}</small>`;
+  }
+
+  function uniqueListFormatter(list: string[]) {
+    if (!list || !list[0]) list = ['(empty)'];
+    const uniq = Array.from(new Set(list));
+    return html`<small>${uniq.join(', ')}</small>`;
+  }
+
+  function timeRemainingLimitFormatter(value: string, row: AssessmentInstanceRow) {
+    return html`
+      ${value}
+      <span>
+        <button
+          class="btn btn-secondary btn-xs ml-1 time-limit-edit-button"
+          id="row${row.assessment_instance_id}PopoverTimeLimit"
+          data-row="${JSON.stringify(row)}"
+        >
+          <i class="bi-pencil-square" aria-hidden="true"></i>
+        </a>
+      </span>
+    `.toString();
+  }
+
+  function detailsLinkFormatter(value: string, row: AssessmentInstanceRow) {
+    const name = assessmentGroupWork ? row.group_name : row.uid;
+
+    let number;
+    if (!assessmentMultipleInstance) {
+      number = row.number === 1 ? '' : `#${row.number}`;
+    }
+    return `<a href="${urlPrefix}/assessment_instance/${value}">${assessmentSetAbbr}${assessmentNumber}${number} for ${name}</a>`;
+  }
+
+  function detailsLinkSorter(
+    valueA: number,
+    valueB: number,
+    rowA: AssessmentInstanceRow,
+    rowB: AssessmentInstanceRow,
+  ) {
+    let nameA: string | null, nameB: string | null, idA, idB;
+    if (assessmentGroupWork) {
+      (nameA = rowA.group_name), (nameB = rowB.group_name);
+      (idA = rowA.group_id ?? ''), (idB = rowB.group_id ?? '');
+    } else {
+      (nameA = rowA.uid), (nameB = rowB.uid);
+      (idA = rowA.user_id ?? ''), (idB = rowB.user_id ?? '');
+    }
+
+    // Compare first by UID/group name, then user/group ID, then
+    // instance number, then by instance ID.
+    let compare = nameA?.localeCompare(nameB ?? '');
+    if (!compare) compare = (parseInt(idA) ?? 0) - (parseInt(idB) ?? 0);
+    if (!compare) compare = (rowA.number ?? 0) - (rowB.number ?? 0);
+    if (!compare) compare = valueA - valueB;
+    return compare;
+  }
+
+  function timeRemainingLimitSorter(
+    valueA: number,
+    valueB: number,
+    rowA: AssessmentInstanceRow,
+    rowB: AssessmentInstanceRow,
+  ) {
+    // Closed assessments are listed first, followed by time limits
+    // ascending, followed by open without a time limit
+    return Number(rowA.open) - Number(rowB.open) || (valueA ?? Infinity) - (valueB ?? Infinity);
+  }
+
+  function actionButtonFormatter(_value: string, row: AssessmentInstanceRow) {
+    const ai_id: number = parseInt(row.assessment_instance_id);
```

Comment:
Why not rely on type inference here?

```suggestion
    const ai_id = parseInt(row.assessment_instance_id);
```

Reply:
Removed the type in [0b3aa97](https://github.com/PrairieLearn/PrairieLearn/pull/9725/commits/0b3aa97bfea69bfe855c4af0ac526d99182d6f6b).